docs/source/devel/design.rst

```diff
@@ -1,19 +1,214 @@
+          - | Role threshold
+            | It uses the role name uppercase
+            | Example: ``ROOT_THRESHOLD``
+        * - ``NUMBER_OF_DELEGATED_BINS``
+          - | ``int``
+            | Number of delegated bins
+          - number of delegated hash bin roles
+        * - ``SIGNING_<ROLE NAME>``
+          - | ``None``
+            | ``<json>``
+          - | ``None``: No pending signature(s)
+            | ``json``: TUF Metadata pending signature
+            | It uses the role name uppercase
+            | Example ``SIGNING_ROOT``
+
+Target Files and Target Roles
+=============================
+
+    * The TUF top Role Targets is a role only for delegation. This role does
+      not register target files (artifacts).
+    * :ref:`devel/design:PostgreSQL` stores the artifacts (``TargetFiles``) and
+      Targets delegated roles.
+    * :ref:`devel/design:Repository Service TUF Worker` manages the
+      :ref:`devel/design:PostgreSQL` database.
+    * :ref:`devel/design:Repository Service TUF Worker` implements and manages
+      the Key Vault and Storage Services.
+
+      - Access to the Key Vault Service is restricted to
+        :ref:`devel/design:Repository Service TUF Worker`
+      - Writing the TUF Metadata in the Storage Service  is limited to
+        :ref:`devel/design:Repository Service TUF Worker`
+      - The Storage Service is the only public data
+
+RSTUF Components Design
+#######################
+
+Repository Service TUF API
+==========================
+
+* Integration (add/remove artifacts)
+* TUF metadata process (bootstrap, sign, update, etc)
+
+.. note::
+    The service can implement other features without interfering with the
+    RSTUF principles and architecture design principles.
+
+`See component development documentation
+<https://repository-service-tuf.readthedocs.io/projects/rstuf-api/en/latest/devel/>`_.
+
+
+Repository Service TUF Worker
+=============================
+
+* Manages the TUF metadata
+* Manages the Key Vault and Storage Services
+
+.. note::
+    The service can implement other features without interfering with the
+    RSTUF principles and architecture design principles.
+
+`See component development documentation
+<https://repository-service-tuf.readthedocs.io/projects/rstuf-worker/en/latest/devel/>`_.
+
+Repository Service TUF CLI
+==========================
+
+* Command Line Interface for the API
+* Guide users in the process
+
+.. note::
+    The service can implement other features without interfering with the
+    RSTUF principles and architecture design principles.
+
+`See component development documentation
+<https://repository-service-tuf.readthedocs.io/projects/rstuf-cli/en/latest/devel/>`_.
+
+
+RSTUF Infrastructure Services Design
+####################################
+
+The Infrastructure Services have key functionality on RSTUF
+
+Message Queue
+=============
+
+* It is a centralized queue
+
+Backend Result
+==============
+
+* It is a centralized backend result for tasks
```

Comment:
Is this a service or what is it?
The description doesn't give much information.

Reply:
I will add more information

Reply:
I added more the context for Celery, included the same to message queue